src/runtime/http.rs

```diff
@@ -54,28 +69,47 @@ impl HttpClient {
         &self,
         method: Method,
         uri: &str,
-        headers: impl IntoIterator<Item = &'a (&'a str, &'a str)>,
-    ) -> reqwest::Result<String> {
-        let text = self.request(method, uri, headers).await?.text().await?;
+        headers: &'a [(&'a str, &'a str)],
+    ) -> Result<String, Box<dyn Error>> {
+        let res = self.request(method, Uri::from_str(uri)?, headers).await?;
 
+        let bytes = body::to_bytes(res.into_body()).await?;
+
+        let text = String::from_utf8(bytes.to_vec())?;
         Ok(text)
     }
 
-    /// Executes an HTTP equest and returns the response.
-    pub(crate) async fn request<'a>(
-        &self,
+    /// Executes an HTTP request and returns the response.
+    pub(crate) fn request<'a>(
+        &'a self,
         method: Method,
-        uri: &str,
-        headers: impl IntoIterator<Item = &'a (&'a str, &'a str)>,
-    ) -> reqwest::Result<Response> {
-        let response = headers
-            .into_iter()
-            .fold(self.inner.request(method, uri), |request, (k, v)| {
-                request.header(*k, *v)
-            })
-            .send()
-            .await?;
+        uri: Uri,
+        headers: &'a [(&'a str, &'a str)],
```

Comment:
Is there a reason you replaced the `IntoIterator` logic? The `Iterator::fold` pattern used previously would probably work here too.

Reply:
Yeah, https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.fold takes `self` as an argument causing the iterator to get consumed. I think this prevents it from being used further in the redirect.